app/actions/remote/preference.ts

```diff
@@ -128,6 +134,87 @@ export const deleteSavedPost = async (serverUrl: string, postId: string) => {
+        if (!channel || !isDMorGM(channel)) {
+            return {};
+        }
+        const userId = await getCurrentUserId(database);
+        const {DIRECT_CHANNEL_SHOW, GROUP_CHANNEL_SHOW} = Preferences.CATEGORIES;
+        const category = channel.type === General.DM_CHANNEL ? DIRECT_CHANNEL_SHOW : GROUP_CHANNEL_SHOW;
+        const name = channel.type === General.DM_CHANNEL ? getUserIdFromChannelName(userId, channel.name) : channelId;
+        const preferences = await queryPreferencesByCategoryAndName(database, category, name).fetch();
+        const visible = getPreferenceAsBool(preferences, category, name, false);
+        if (visible) {
+            return {};
+        }
+
+        const prefs: PreferenceType[] = [
+            {
+                user_id: userId,
+                category,
+                name,
+                value: 'true',
+            },
+            {
+                user_id: userId,
+                category: Preferences.CATEGORIES.CHANNEL_OPEN_TIME,
+                name: channel.id,
+                value: Date.now().toString(),
+            },
+        ];
+
+        return savePreference(serverUrl, prefs);
+    } catch (error) {
+        forceLogoutIfNecessary(serverUrl, error);
+        return {error};
+    }
+};
+
+export const openAllUnreadChannels = async (serverUrl: string) => {
+    try {
+        const {database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+        const channels = await queryAllUnreadDMsAndGMsIds(database).fetch();
+        const userId = await getCurrentUserId(database);
+        const {DIRECT_CHANNEL_SHOW, GROUP_CHANNEL_SHOW} = Preferences.CATEGORIES;
+
+        const directChannelShowPreferences = await queryPreferencesByCategoryAndName(database, DIRECT_CHANNEL_SHOW).fetch();
+        const groupChannelShowPreferences = await queryPreferencesByCategoryAndName(database, GROUP_CHANNEL_SHOW).fetch();
+        const showPreferences = directChannelShowPreferences.concat(groupChannelShowPreferences);
+        const prefs: PreferenceType[] = [];
+        for (const channel of channels) {
+            const category = channel.type === General.DM_CHANNEL ? DIRECT_CHANNEL_SHOW : GROUP_CHANNEL_SHOW;
+            const name = channel.type === General.DM_CHANNEL ? getUserIdFromChannelName(userId, channel.name) : channel.id;
+            const visible = getPreferenceAsBool(showPreferences, category, name, false);
+            if (visible) {
+                continue;
+            }
+
+            prefs.push(
+                {
+                    user_id: userId,
+                    category,
+                    name,
+                    value: 'true',
+                },
+                {
+                    user_id: userId,
+                    category: Preferences.CATEGORIES.CHANNEL_OPEN_TIME,
+                    name: channel.id,
+                    value: Date.now().toString(),
+                },
+            );
```

Comment:
this is almost identical to `openChannelIfNeeded` perhaps create a function that you can call from both actions?